Update facility download

TaiWilkin opened the pull request:
## Overview

Updates the facility download format to split facility list items to
separate rows, with each facility having one base row and 0 or more
facility list item rows. In cases where a claim exists, the base row
is listed as contributed by the claimant; in other cases, the facility
created_from contributor is cited.

The facility CSV/XLSX download has become increasingly more complex
over time. Previously we used the facility search results
directly to download the list, but the additional data required to
render the download began to involve a lot of overhead, and we switched
to only downloading the more detailed version of the list when the
download was requested. With the latest changes to the format, the
required data was expected to again increase, in addition to the
complexity of the formatting increasing.

To simplify the process and decrease the amount of data fetched, the
formatting of the data into a CSV/XLSX-friendly format has been moved
to the backend.

Connects #1752 

## Demo

### CSV and XLSX files
[embed.csv](https://github.com/open-apparel-registry/open-apparel-registry/files/9153861/embed.csv)
[new_download.xlsx](https://github.com/open-apparel-registry/open-apparel-registry/files/9153863/new_download.xlsx)
[new_download.csv](https://github.com/open-apparel-registry/open-apparel-registry/files/9153865/new_download.csv)

### Timing

I timed the logDownload function locally while downloading 1108 facilities. This is a very rough estimate of timing, because we have significantly more list items per facility on staging and production, but the results are encouraging.

Time to download 1108 facilities on this branch: 36664.72119140625 ms (36.66 seconds)
Time to download 1108 facilities on develop: 97642.63012695312 ms (97.64 seconds)

## Notes

The previous front-end implementation of this included some sections which were wrapped in flags (embedded map, extended fields, PPE, claims, and closure status). It doesn't seem likely that we will remove embedded map, extended fields, claims, or closure status in the future, so I didn't wrap these portions of the code in flags. PPE is in the process of being phased out, so I didn't include PPE columns for download at all. 

I have it set up to filter out list items that are inactive, private, or have a status other than AUTOMATIC, CONFIRMED, or MERGED. Should some of these items instead be included but have the contributor anonymized? Are there are instances where the contributor should be anonymized?

## Testing Instructions

* Have the following types of facilities in your local database, each submitted at least once by the same contributor:
     - A claimed facility, unapproved
     - A claimed facility, approved
     - A facility list item with extended fields
     - A facility list item with nonstandard/contributor fields
     - An inactive or private facility list item
* Download search results including the given facilities as a CSV
* Download the same as XLSX
* Open the embedded map for the contributor and download the same facilities
* Confirm that each download contains the expected data and formatting
* View the new tests and confirm that the expected results for each test match your understanding of the expected download formatting

## Checklist

- [x] `fixup!` commits have been squashed
- [ ] CI passes after rebase
- [x] CHANGELOG.md updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines
- [x] This PR is targeted at the correct branch (`develop` vs. `ogr/develop`)
- [ ] If this PR applies to both OAR and OGR a companion PR has been created

jwalgran reviewed (changes requested):
Nice job working through a detailed reworking of a complex feature. I appreciate
the comprehensive test cases. 

Is there a reason this is targeted at OAR instead of OS Hub? I had imagined that we would only deploy this reworking of the downloads to the OS Hub beta.

> I have it set up to filter out list items that are inactive, private, or have a status other than AUTOMATIC, CONFIRMED, or MERGED. Should some of these items instead be included but have the contributor anonymized? Are there are instances where the contributor should be anonymized?

I do think it is correct to filter out items with a status other than AUTOMATIC,
CONTRIBUTOR, or MERGED but I think we do think we want to include rows with
anonymized contributor details rather then excluding them. The facility
details page will show all the extended fields from an inactive or private
source so for consistency it makes sense to also include them in the download.

Related to anonymous contributor names, I found a bug where we are showing a
contributor name when it should be hidden

- POST this facility
```
{
    "country": "US",
    "name": "Azavea",
    "address": "990 Spring Garden St., Philadelphia PA 19123",
    "facility_type_processing_type": "stiching",
    "parent_company": "foo",
    "product_type": "shoes"
}
```
- Browse http://localhost:8081/admin/api/source/ , edit the most recent source, and make it inactive
- Search for Azavea, go to the details page, and verify the the contributor name
is anonymized
- Download the results. The full name appears in the contributor column

<img width="512" alt="Screen Shot 2022-07-20 at 9 05 46 PM" src="https://user-images.githubusercontent.com/17363/180127870-8ae42d22-9387-420d-aed1-441b7dad5a51.png">

TaiWilkin commented on src/django/api/views.py:
```diff
@@ -2560,6 +2562,61 @@ def link(self, request, pk=None):
+        ### Sample Response
+            {
+
+            }
```
Removed

TaiWilkin commented on src/django/api/csv_download.py:
```diff
@@ -0,0 +1,97 @@
+
+def combine_raw_values(new_values, old_values):
+    old_set = set()
+    if len(old_values) is not 0:
```
Updated

TaiWilkin commented on src/django/api/csv_download.py:
```diff
@@ -0,0 +1,97 @@
+
+
+def format_download_date(date):
+    return date.strftime("%d-%b-%y")
```
It was formatted that way in the demo download from the OAR team, but I think it does make more sense to use `YYYY-MM-DD` since we use that elsewhere. Updated.

TaiWilkin commented:
I rebased onto OSH and force-pushed the changes (adding sectors to facilities, mostly) needed for that.
The other changes are contained in the fixup commits ([69f7371](https://github.com/open-apparel-registry/open-apparel-registry/pull/1981/commits/69f737136f5d415cee1f08773fcf1912cf5a5e85) and [cc4cf63](https://github.com/open-apparel-registry/open-apparel-registry/pull/1981/commits/cc4cf630b824ba953cf686923ab77a20abde21e6)) to keep them separated out for reviewing.
I updated the follow-up rows to only exclude invalid statuses, and to include inactive/private sources and matches. In relation to that, I updated the row rendering to anonymize contributors for private/inactive sources and matches. I also added your steps to replicate the bug as a new test to make sure it stays fixed.
This is ready for another look.

jwalgran reviewed (approved):
Changes look good and the anonymization works as expected in my tests. Thank you.

TaiWilkin commented:
Thank you for reviewing!

mariel-oar commented:
@jwalgran @obrienad , can you confirm if this was pushed to OAR at some point? (trying to confirm that this new download format was live on OAR? before the freeze
)

jwalgran commented:
No. We only modified the download format on the OS Hub
